client/app/components/account_config_main.coffee

```diff
@@ -268,214 +179,85 @@ module.exports = AccountConfigMain = React.createClass
-                else if server.socketType is 'STARTTLS'
-                    infos.imapSSL = false
-                    infos.imapTLS = true
-
-                else if server.socketType is 'plain'
-                    infos.imapSSL = false
-                    infos.imapTLS = false
-
-            if server.type is 'smtp' and not infos.smtpServer?
-                infos.smtpServer = server.hostname
-                infos.smtpPort = server.port
-
-                if server.socketType is 'SSL'
-                    infos.smtpSSL = true
-                    infos.smtpTLS = false
-
-                else if server.socketType is 'STARTTLS'
-                    infos.smtpSSL = false
-                    infos.smtpTLS = true
-
-                else if server.socketType is 'plain'
-                    infos.smtpSSL = false
-                    infos.smtpTLS = false
-
-        # Set default values if providers didn't give required infos.
-
-        unless infos.imapServer?
-            infos.imapServer = ''
-            infos.imapPort   = '993'
-
-        unless infos.smtpServer?
-            infos.smtpServer = ''
-            infos.smtpPort   = '465'
-
-        unless infos.imapSSL
-            switch infos.imapPort
-                when '993'
-                    infos.imapSSL = true
-                    infos.imapTLS = false
-                else
-                    infos.imapSSL = false
-                    infos.imapTLS = false
-
-        unless infos.smtpSSL
-            switch infos.smtpPort
-                when '465'
-                    infos.smtpSSL = true
-                    infos.smtpTLS = false
-                when '587'
-                    infos.smtpSSL = false
-                    infos.smtpTLS = true
-                else
-                    infos.smtpSSL = false
-                    infos.smtpTLS = false
-
-        # Display gmail warning if selected provider is Gmail.
-        isGmail = infos.imapServer is 'imap.googlemail.com'
-        @setState displayGMAILSecurity: isGmail
-
-        # Apply built values to current state
-        @state[key].requestChange val for key, val of infos
-
-
-    # Configure port automatically depending on selected paramerters.
-    _onServerParam: (target, server, type) ->
-
-        # port has been set manually, don't update it
-        if not((server is 'imap' and @state.imapManualPort) or
-        (server is 'smtp' and @state.smtpManualPort))
-
-            if server is 'smtp'
-
-                if type is 'ssl' and target.checked
-                    @setState smtpPort: 465
-
-                else if type is 'tls' and target.checked
-                    @setState smtpPort: 587
-
+    doDiscovery: (domain) ->
+        if domain? and domain.length > 3 and domain isnt @_lastDiscovered
+            console.log domain, @discoverTimeout
+            if @discoverTimeout
+                @nextDiscover = domain
             else
+                @discoverTimeout = setTimeout (=> @doDiscoveryNow domain), 2000
```

Comment:
I tried a bit, but had trouble interfacing it with React autobinding. This code will need moving to a proper actions/store anyway. I will try again to use _.throttle or _.debounce at this point.